mobile/native/src/ln_dlc/channel_status.rs

```diff
@@ -0,0 +1,197 @@
+use crate::db::get_all_non_pending_channels;
+use crate::event;
+use crate::ln_dlc::node::Node;
+use anyhow::Result;
+use ln_dlc_node::channel::Channel;
+use ln_dlc_node::channel::ChannelState;
+use ln_dlc_node::node::rust_dlc_manager::subchannel::SubChannel;
+use std::borrow::Borrow;
+use std::time::Duration;
+
+const UPDATE_CHANNEL_STATUS_INTERVAL: Duration = Duration::from_secs(5);
+
+/// The status of the app channels.
+#[derive(Debug, Clone, Copy)]
+pub enum ChannelStatus {
+    /// No channel is open.
```

Comment:
have you considered having an `Unknown` channel status as the enum variant?

That's how I went with `ServiceStatus`. I'm not sure if any of the 2 options is better, but it would be good to align on one approach.

Personally, I prefer not sending `null` over (although apparently we have null safety in Dart3), but this might be my bias after years of pain in C/C++ world. It also ensures we can stringify enums nicer etc, as we can't forget about that wariant in these places  - with `null`, things can get a bit ad-hoc and inconsistent as they might be done in multiple places.

Happy to be convinced otherwise though!

Reply:
I definitely considered it! Since you are suggesting it too I'm happy to try to model it differently.